### ISSUE TITLE
Failed Tweets is always retried with no exhaustion limit

### ISSUE THREAD

**Issue by chez14** [direbus/daily-watori#28]
Based on log, it appears that DailyWatori tried to retweet the same tweet ID for...ever without exhaustion limit.

```
$ docker-compose exec app tail bot.log
[1615507200247] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358097637912707077
[1615507200318] INFO (daily-watori/24 on 8c1329a4e332): Fetched 9 tweets from 21 users
[1615536000002] INFO (daily-watori/24 on 8c1329a4e332): Scheduled retweet started
[1615536000208] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358361630724136962
[1615536000220] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358097637912707077
[1615536000232] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358097690966466561
[1615564800001] INFO (daily-watori/24 on 8c1329a4e332): Scheduled retweet started
[1615564800217] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358361630724136962
[1615564800234] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358097637912707077
[1615564800245] ERROR (daily-watori/24 on 8c1329a4e332): Failed to retweet a tweet with id 1358097690966466561
```

**Comment by Namchee** [comment]
Hmm, I think it's a good idea to log the reason why retweet fails. wydt, @chez14? 

As for this issue, what's the expected behavior when retweet fails?

**Comment by chez14** [comment]
I think thats good.

I'm expecting that at least we flag the queue as failed. I'm thinking about adding a new table to log the error details, but that is optional.

All what I want is, there's a way to skip the tweet that unable to be retweeted.

---

I'm actually working on this issue, and having problem with #29, and now been reverted on #29.

**Comment by chez14** [comment]
To be honest, i wanted to try the API call for 3 times, but that kind of approach wont work... right? Unless there's a connection error, or similar exceptions.

But well... that's another story to think about when I can't sleep. So I'm going to skip this and revisit later.

**Comment by repurika** [comment]
:rocket: Issue was released in [`v0.3.0`](https://github.com/direbus/daily-watori/releases/tag/v0.3.0) :rocket:
